src/catalog/backends/common/src/main/java/org/geoserver/cloud/autoconfigure/catalog/backend/core/RemoteEventResourcePoolCleanupUpAutoConfiguration.java

```diff
@@ -20,14 +21,14 @@
  * @since 1.0
  */
 @AutoConfiguration
-@ConditionalOnClass(InfoEvent.class)
+@ConditionalOnClass(value = {InfoEvent.class, LifecycleEvent.class})
 @ConditionalOnCatalogEvents
-public class RemoteEventResourcePoolCleaupUpAutoConfiguration {
```

Comment:
Thanks for fixing the typo.
Note this `@AutoConfiguration` class is specifically for `RemoteEventResourcePoolProcessor`, which in turn is used to clean up the `Catalog`'s `ResourcePool` caches. So `RemoteEventResourcePoolProcessor` is not the right place to handle the `LyfeCycleEvents`.

I'd suggest to create a new `@AutoConfiguration` class, say `LyfeCycleEventsAutoConfiguration` (remember to register it in `src/main/resources/META-INF/spring.factories`) that contributes a new service bean to handle the life cycle events in an appropriate package (note `RemoteEventResourcePoolProcessor` is in `org.geoserver.cloud.event.remote.resourcepool` which also has nothing to do with life cycle events).

Reply:
Done in 33c46d24e75dd4a50af3d908695a1e739d75ce4b